libfoundation/src/foundation-number.cpp

```diff
@@ -207,6 +306,73 @@ bool MCNumberParseUnicodeChars(const unichar_t *p_chars, uindex_t p_char_count,
 
 ////////////////////////////////////////////////////////////////////////////////
 
+bool MCNumberFormat(MCNumberRef self, MCStringRef& r_string)
+{
+    switch(__MCNumberGetRepType(self))
+    {
+        case __kMCNumberRepSignedInteger:
+            return MCStringFormat(r_string, "%d", self -> integer);
+        case __kMCNumberRepUnsignedInteger:
+            return MCStringFormat(r_string, "%u", self -> unsigned_integer);
+        case __kMCNumberRepDouble:
+            return MCStringFormat(r_string, "%lf", self -> real);
+        default:
+            MCUnreachable();
+            break;
+    }
+
+    return false;
+}
+
+bool MCNumberTryToParse(MCStringRef p_string, MCNumberRef& r_number)
```

Comment:
How about:
```
bool MCNumberTryToParseRange(MCStringRef p_string, MCRange p_range, MCRange *p_out_range, MCNumberRef& r_number)
```

This attempts to parse 'p_range' of p_string as a number. If 'p_out_range' is specified then it will return the range of characters that were parsed as the number without an end-boundary check. If 'p_out_range' is nil, then a end-boundary check is done.